### ISSUE TITLE
console - check password strength on server side

### ISSUE THREAD

**Issue by severo** [georchestra/georchestra#234]
At user creation, or password modification, the strength of the password is tested on client side (javascript), but not on server side, allowing to enter weak passwords.

Note that the tests should be done on all the classes involved in password creation or password modification.

**Comment by landryb** [comment]
resurrecting this dead ticket, i think we need a allowUserToChoosePasswd boolean somewhere, and if false remove the field on the account creation page & generate strong random passwords and send them to the user upon validation. Letting the user to choose its password, even if there's a hint that it's weak/not great, has always been awkward......

**Comment by severo** [comment]
I disagree... The platform should never know a password, at most its salted hash. Or maybe I missed something?

**Comment by landryb** [comment]
The platform doesn't have to *remember* the password of course, just creating a safe one. 
It's 2019 and when prompted for a password, ppl will still use 12345678 or their first name even if there's a red banner saying it's insecure. Providing a safe one by default, and the possibility to reset it if lost/change it ensures at least a strong one is used - because ppl never change it (and write it on a postit, but that's a different issue).

Another option is to run john the ripper on the ldif hashes, and lock the accounts where the passwords are too easily guessable.. with a notification to the user.

**Comment by severo** [comment]
For trust, I think the platform should not even create the password, because the user could suspect it to keep a copy, or log it, or...
It would be better, imho, to work globally on the authentication mechanisms, for example allowing other mechanisms than LDAP, or allowing 2-factor authentication, etc.

And we could add the option of checking the passwords with jack the ripper in the documentation (or even as a functionnality inside the console, but maybe it's overkill).

**Comment by severo** [comment]
(and https://haveibeenpwned.com/API/v2)
